lib/matplotlib/contour.py

```diff
@@ -1532,7 +1542,7 @@ def _check_xyz(self, args, kwargs):
         Exception class (here and elsewhere).
         """
         x, y = args[:2]
-        self.ax._process_unit_info(xdata=x, ydata=y, kwargs=kwargs)
+        kwargs = self.ax._process_unit_info(xdata=x, ydata=y, kwargs=kwargs)
```

Comment:
I would suggest renaming the method to `_pop_and_process_unit_info` (`_pop_and_apply_unit_info` to save two characters) for clarity -- sneakily modifying a mutable argument usually leads to a lot of head-scratching.